avocado/core/jobdata.py

```diff
@@ -203,7 +207,9 @@ def get_id(path, jobid):
     Gets the full Job ID using the results directory path and a partial
     Job ID or the string 'latest'.
     """
-    if jobid == 'latest':
+    if os.path.isdir(jobid) or os.path.isfile(jobid):
+        jobid = ''
```

Comment:
I find this way hackish, but I already mentioned this in the previous commit... (could be fixed later)